lib/helpers/dotnetutils.js

```diff
@@ -0,0 +1,132 @@
+/**
```

Comment:
These functions were vibe-coded using qwen3-coder. I first gave a simple test case to generate some logic. Then I repeatedly kept asking for more complex test cases and refactoring till we reached a point where we both concluded that the regexes and tests were good enough. It took around two hours of chat!